app/controllers/api/v1x0/mixins/rbac_mixin.rb

```diff
@@ -2,21 +2,26 @@ module Api
   module V1x0
     module Mixins
       module RBACMixin
-
         def write_access_check
-          access_check('write')
+          resource_check('write')
         end
 
         def read_access_check
-          access_check('read')
+          resource_check('read')
         end
 
-        def access_check(verb)
+        def resource_check(verb, id = params[:id], clazz = controller_name.classify.constantize)
```

Comment:
@lindgrenj6 do we have to pass the class name, can't it figure out automatically or do you need it for different resources

Reply:
Yes, for the `PortfolioItem` copy I need to check if we have write access to Portfolios since we're writing into a different portfolio. By default it'll go off of the controller, but because I'm in the PortfolioItem controller I need the ability to pass in a class to run the resource_check on.

By default it's still figuring out which resource to check based on the controller - this just gives a way to override it in case you want to check access to something that isn't the controller you're curren't in.